### ISSUE TITLE
[IE 0.12-78] Skyhook Permanent Disconnect - "Invalid move player packet received"

### ISSUE THREAD

**Issue by seriallos** [BluSunrize/ImmersiveEngineering#2872]
### Description of the issue:

I'm encountering several stability issues with skyhooks on a multiplayer server.

Was trying to use skyhook on an insulated LV wire between two powered lanterns on top of wooden posts (the IE treated wood ones) when I got my player into a bad state. Server disconnects me with "Invalid move player packet received" whenever I try to log into the server.

It took me a couple tries to get on the line and I was able to briefly attach successfully. It might be that attaching close to one of the lanterns triggered the disconnect/invalid position.

Update - edited my player to get Y coordinate back into the world and was able to play around some more. One misunderstanding I had was that you have to hold down RMB to use the skyhook properly. I was clicking and releasing quickly when I encountered the issue.

### Crashlog:

First disconnect (note that there are no messages between crafting the skyhook and then actually trying to use it 30 minutes later):

```
Mar 12 23:31:43 mc-server supervisord: mc-server [23:31:43] [Server thread/INFO]: sollaires has made the advancement [Booker, catch!]
Mar 13 00:05:06 mc-server supervisord: mc-server [00:05:06] [Server thread/INFO]: sollaires lost connection: Invalid move player packet received
Mar 13 00:05:06 mc-server supervisord: mc-server [00:05:06] [Server thread/INFO]: sollaires left the game
```

Server logs on connection (Maybe note the `NaN` in the player position?)

```
Mar 13 00:27:51 mc-server supervisord: mc-server [00:27:51] [Server thread/INFO]: sollaires[/X.X.X.X:58752] logged in with entity id 420 at (-81.94579111426903, NaN, 646.5)
Mar 13 00:27:51 mc-server supervisord: mc-server [00:27:51] [Server thread/INFO]: sollaires joined the game
Mar 13 00:27:51 mc-server supervisord: mc-server [00:27:51] [Server thread/INFO] [journeymap]: sollaires joining dimension 0
Mar 13 00:27:51 mc-server supervisord: mc-server [00:27:51] [Server thread/INFO] [journeymap]: Sending log in packet.
Mar 13 00:27:53 mc-server supervisord: mc-server [00:27:53] [Server thread/INFO]: sollaires lost connection: Invalid move player packet received
Mar 13 00:27:53 mc-server supervisord: mc-server [00:27:53] [Server thread/INFO]: sollaires left the game
```

Some stack trace logs from the server when just playing around with the skyhook resulting in a different permanent disconnect loop:

https://pastebin.com/X31Xm4Kj

### Versions & Modlist

Same mod versions from my other recently created issue: https://pastebin.com/zv75DEnM (this is not a crash log for this issue)

**Comment by seriallos** [comment]
One note that might be useful in a README or somewhere is that `allow-flight=true` is probably desired for an IE server. I haven't been able to repro the invalid position again but I was able to get the server to kick me for not allowing flying.

**Comment by seriallos** [comment]
Playing around some more and I've gotten myself into a pickle I can't solve. Getting disconnected every time I try to connect with this showing up in the server logs:

https://pastebin.com/X31Xm4Kj

Edited `Pos` in my player .dat file but I'm still crashing on login. Any other data on my player I can edit to get out of the attached mode or maybe solve the disconnect loop?

**Comment by seriallos** [comment]
One last note for now - deleting the `RootVehicle` entry in my .dat file let me log back in.

**Comment by lgthibault** [comment]
Trying to repro some crazy right-clicking to generate NaNs.

I looked at https://pastebin.com/X31Xm4Kj
The NPE at MessageSkyhookSync.java:63 is likely a side-effect of losing the connection (connection.writeToNBT())
The NPE at EntitySkylineHook.java:103 possibly also the same ((this.ticksExisted==1&&!world.isRemote))
They repeat 16 times each and there is nothing more there.

RootVehicle doesn't light up any class in the IE source tree.

**Comment by ghostshadow** [comment]
I experienced something simmilar, trying to use the skyhook on a rope between two structural connectors. The client as well as the server crashed and had to be killed.
It might have something to do, with the rope theoretically pulling the player through a block ... (at least in my case, the rope has a rather extreme slope, which gets it just above the edge of the block)

**Comment by malte0811** [comment]
This is probably fixed in the next IE build (81), see #2937. I'm going to close this, please put a comment here if this happens again with IE 81 or maven build 363+. The existing broken players need to be fixed manually as I said in the other issue.
